src/Features/CSharp/Portable/SignatureHelp/SignatureHelpUtilities.cs

```diff
@@ -46,18 +46,7 @@ internal static class SignatureHelpUtilities
                     ? argument.NameColon.Name.Identifier.ValueText
                     : argument.NameEquals?.Name.Identifier.ValueText);
 
-        internal static SignatureHelpState? GetSignatureHelpState(BaseArgumentListSyntax argumentList, int position, int parameterIndex)
-        {
-            var result = GetSignatureHelpState(argumentList, position);
-            if (result is not null && parameterIndex >= 0)
-            {
-                result.ArgumentIndex = parameterIndex;
```

Comment:
this was the place where we unilaterally override the index.  taht was the problem.